geomstats/geometry/klein_bottle.py

```diff
@@ -0,0 +1,392 @@
+        """
+        if Metric is None:
+            Metric = self._default_metric()
+        self.metric = Metric(self)
+
+        return self.metric
+
+    def random_point(self, n_samples=1, bound=None):
+        """Uniformly sample points on the manifold.
+
+        Parameters
+        ----------
+        n_samples : int
+            Number of samples.
+            Optional, default: 1.
+        bound : unused
+
+        Returns
+        -------
+        samples : array-like, shape=[n_samples, 2]
+            Points sampled on the manifold.
+        """
+        samples = gs.random.uniform(size=(n_samples, 2))
+        if n_samples == 1:
+            samples = gs.squeeze(samples, axis=0)
+        return samples
+
+    def to_tangent(self, vector, base_point=None):
+        """Project a vector to a tangent space of the manifold.
+
+        In this case returns the vector itself.
+
+        Parameters
+        ----------
+        vector : array-like, shape=[..., dim]
+            Vector.
+        base_point : unused
+
+        Returns
+        -------
+        tangent_vec : array-like, shape=[..., dim]
+            Tangent vector at base point.
+        """
+        if not gs.all(self.is_tangent(vector)):
+            raise ValueError("Cannot handle non tangent vectors")
+
+        return gs.copy(vector)
+
+    def is_tangent(self, vector, base_point=None, atol=gs.atol):
+        r"""Evaluate if the point belongs to :math:`\mathbb R^2`.
+
+        This method checks the shape of the input point.
+
+        Parameters
+        ----------
+        vector : array-like, shape=[..., dim]
+            Point to test.
+        base_point: unused here
+        atol : float
+            Unused here.
+
+        Returns
+        -------
+        belongs : array-like, shape=[...,]
+            Boolean evaluating if point belongs to the tangent space.
+        """
+        # TODO: use errors after merge
+        minimal_ndim = len(self.shape)
+        correct_shape = vector.shape[-minimal_ndim:] == self.shape
+        if correct_shape:
+            if vector.ndim > minimal_ndim:
+                return gs.ones(*vector.shape[:-minimal_ndim])
+            else:
+                return correct_shape
+        raise ValueError(
+            f"Wrong shape: shape of vector should end with {self.shape} "
+            f"but is {vector.shape}"
+        )
+
+    def belongs(self, point, atol=gs.atol):
+        """Evaluate if the point belongs to the set [0,1]^2.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., dim]
+            Point to test.
+        atol : float
+            Unused here.
+
+        Returns
+        -------
+        belongs : array-like, shape=[...,]
+            Boolean evaluating if point belongs to the space.
+        """
+        return gs.all(gs.logical_and(point >= 0, point <= 1), axis=-1)
+
+    @staticmethod
+    def equivalent(point1, point2, atol=gs.atol):
```

Comment:
rename:
point1 --> point_a
point2 --> point_b
for consistency with the current API